### ISSUE TITLE
Geometry compression for face data by cells in MappingInfo

### ISSUE THREAD

**Issue by kronbichler** [dealii/dealii#10763]
As observed in #10762 we need to be careful with the outside storage because the exterior cells could be non-affine (and thus, not amenable to the Jacobian compression) whereas the interior cells are affine. The latter currently control the memory to be allocated in this code:
https://github.com/dealii/dealii/blob/cf555a201b5a9f44a28b306433fb7df99763197e/include/deal.II/matrix_free/mapping_info.templates.h#L2890-L2913
We need to separately look into those. As to a mesh to check this behavior, we can use the mesh created by
https://github.com/dealii/dealii/blob/master/tests/matrix_free/create_mesh.h

**Comment by peterrum** [comment]
@kronbichler  I don't think we will be able to implement a fix till the release. Postponing.

**Comment by kronbichler** [comment]
We won't have this until the 9.4 release, either, even though we made progress on some related topics. Adjusting target milestone.
